ws/adapter.go

```diff
@@ -0,0 +1,89 @@
+package ws
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gorilla/websocket"
+	"github.com/mainflux/mainflux"
+)
+
+const protocol = "ws"
+
+var (
+	_ Service = (*adapterService)(nil)
+
+	// ErrFailedMessagePublish indicates that message publishing failed.
+	ErrFailedMessagePublish = errors.New("failed to publish message")
+	// ErrFailedMessageBroadcast indicates that message broadcast failed.
+	ErrFailedMessageBroadcast = errors.New("failed to broadcast message")
+)
+
+// Service contains publish and subscribe methods necessary for
+// message transfer.
+type Service interface {
```

Comment:
It feels like we should move `Broadcast` to the `MessagePublisher` interface. Also, it makes sense to create `MessageSubscriber` interface as well.